openedx/features/badging/helpers.py

```diff
@@ -0,0 +1,31 @@
+import copy
+
+
+def create_trophycase_data(trophycase_dict, badge_type, all_badges, earned_user_badges, enrolled_courses_data):
```

Comment:
To me this method is confusing. Half of the work we're doing in the view even though we're only using the extracted data in this method. If this whole code block is to be used again lets shift the queries in the model into this method. If this isn't to be used again lets shift this to the view.
Moreover lets use a more specific name for this method as to what it does